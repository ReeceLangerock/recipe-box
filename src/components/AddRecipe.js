import React from 'react'
import styled from 'styled-components'
import * as actions from '../redux/actions/actions'
import { connect } from 'react-redux'

const Container = styled.div`
  width: calc(100% * (1/4) - 20px - 1px);
  min-width: 200px;

  margin: 10px 5px 10px 5px;
  box-sizing: border-box;
  overflow: hidden;
  text-align: left;
  transition: all 0.75s ease-in-out;
  max-height: 60px;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  border-radius: 3px;

  &:hover {
    transition: all 0.75s ease-in-out;

    box-shadow: 0 10px 20px rgba(155, 197, 61, 0.29), 0 6px 6px rgba(155, 197, 61, 0.23);
  }
  @media (max-width: 920px) {
    width: calc(100% * (1/3) - 10px - 2px);
  }

  @media (max-width: 700px) {
    width: calc(100% * (1/2) - 10px - 2px);
  }
  @media (max-width: 500px) {
    width: calc(100% - 10px - 1px);
  }
`

const Name = styled.div`
  font-size: 1.5rem;
  display: flex;
  justify-content: space-between;
`

const Header = styled.div`
  align-items: center;
  padding: 16px 10px 16px 10px;
  transition: all 0.75s ease-in-out;
  color: white;
  background-color: rgba(155, 197, 61, 1);
  cursor: pointer;
`

export class AddRecipe extends React.Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    this.props.setAddModal('add')
  }

  render () {
    return (
      <Container onClick={this.handleClick}>
        <Header>
          <Name>
            <span>Add Recipe</span>
            <span>
              <i className='fa fa-plus' aria-hidden='false' />
            </span>
          </Name>
        </Header>
      </Container>
    )
  }
}
export default connect(null, actions)(AddRecipe)
