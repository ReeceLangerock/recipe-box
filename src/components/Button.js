import React from "react";
import styled from "styled-components";

export class Button extends React.Component {
  render() {
    return <div>Button</div>;
  }
}
export default Button;
