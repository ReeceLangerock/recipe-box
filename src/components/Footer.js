import React from "react";
import styled from "styled-components";

const FooterStyled = styled.div`
  a {
  }
`;

const I = styled.i`
  margin-left: 10px;
  color: black;
  cursor: pointer;
`;

export class Footer extends React.Component {
  handleIconClick(link) {
    window.open("https://github.com/ReeceLangerock/camper-leaderboard", "_blank");
  }
  render() {
    const currentYear = new Date().getFullYear();

    return (
      <FooterStyled className="Footer">
        Â© {currentYear}{" "}
        <a href="http://reecelangerock.com" rel="noopener noreferrer" target="_blank">
          Reece Langerock
        </a>
        <I className={`fa fa-github`} aria-hidden="false" onClick={() => this.handleIconClick()} />
      </FooterStyled>
    );
  }
}
export default Footer;
