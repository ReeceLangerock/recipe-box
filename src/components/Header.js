import React from "react";
import styled from "styled-components";
import { injectGlobal } from "styled-components";
import moonFlower from "./../fonts/Moon Flower.ttf";

injectGlobal`
  @font-face {
    font-family: 'Moon Flower';
    src: url('${moonFlower}');
  }
  

`;

const StyledHeader = styled.div`
  font-size: 9rem;
  color: rgb(155, 197, 61);
  margin-top: 20px;
  margin-bottom: 20px;
  font-family: "Moon Flower";
  transtion: 0.75s ease-in;

  @media (max-width: 500px) {
    font-size: 5rem;
    transtion: 0.75s ease-in;
  }
`;

export class Header extends React.Component {
  render() {
    return <StyledHeader>Recipe Box</StyledHeader>;
  }
}
export default Header;
