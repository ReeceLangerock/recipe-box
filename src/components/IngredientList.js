import React from "react";
import styled from "styled-components";

export class IngredientList extends React.Component {
  render() {
    return <div>IngredientList</div>;
  }
}
export default IngredientList;
