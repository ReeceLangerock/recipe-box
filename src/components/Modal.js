import React from "react";
import styled from "styled-components";
import { connect } from "react-redux";
import * as actions from "../redux/actions/actions";

import Button from "./Button";

const ModalContainer = styled.div`
  position: absolute;
  z-index: 10;
  min-width: 50%;
  border: 1px solid rgba(155, 197, 61, 0.5);
  border-radius: 5px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
  padding: 15px;
  top: 33%;
  margin: 0 auto;
  transition: all 0.75s ease-in-out;
  left: ${props => (props.active ? "50%" : "200%")};
  transform: translate(-50%, 0);
  text-align: left;
  opacity: 1;
  color: #2f2f2f;
  background: white;
  box-sizing: border-box;

  &:hover {
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.29), 0 6px 6px rgba(0, 0, 0, 0.23);
  }
  textarea,
  input {
    position: relative;
    width: calc(100% - 5px);
    border: 2px solid rgba(155, 197, 61, 0.75);
    border-radius: 3px;
    outline-color: rgba(155, 197, 61, 1);
    padding: 2px;
  }
`;

const Label = styled.div`
  font-size: 1.25rem;
  font-weight: bold;
  margin-top: 10px;
  margin-bottom: 3px;
`;

const Form = styled.div`
  margin-bottom: 20px;
  margin-top: 10px;
`;

const Header = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 1.4rem;
  font-weight: none;
  margin: 5px 0px 5px 0px;
`;

const XSpan = styled.span`
  cursor: pointer;
  z-index: 11;
`;

export class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.handleInputChange = this.handleInputChange.bind(this);
    let index = this.props.editableRecipeIndex;
    let name = index >= 0 ? this.props.recipes[index].name : "";
    let ingredients = index >= 0 ? this.props.recipes[index].ingredients : "";

    this.state = {
      name: name,
      ingredients: ingredients
    };
  }

  componentWillReceiveProps(newProps) {
    console.log(newProps);
    let index = newProps.editableRecipeIndex;
    let name = index >= 0 ? newProps.recipes[index].name : "";
    let ingredients = index >= 0 ? newProps.recipes[index].ingredients : "";
    this.setState({
      name: name,
      ingredients: ingredients
    });
  }

  handleInputChange(event) {
    const field = event.target.name;
    const value = event.target.value;
    console.log(value);
    this.setState({
      [field]: value
    });
  }

  render() {
    return (
      <ModalContainer active={this.props.activeModal !== ""}>
        <Header>
          <span>{this.props.modalHeader}</span>
          <XSpan onClick={this.props.clearModal}>X</XSpan>
        </Header>

        <hr />
        <Form>
          <Label>Recipe</Label>
          <input name={"name"} placeholder={"Recipe name"} value={this.state.name} onChange={this.handleInputChange} />
          <Label>Ingredients</Label>

          <textarea name={"ingredients"} placeholder={"Enter Ingredients, seperate by commas"} value={this.state.ingredients} onChange={this.handleInputChange} />
        </Form>
        <div>
          <Button
            isDisabled={this.state.name === "" || this.state.ingredients === ""}
            buttonType={"confirm"}
            buttonText={this.props.modalButtonText}
            buttonMethod={this.props.modalButtonMethod}
            payload={[this.state.name, this.state.ingredients, this.props.editableRecipeIndex]}
          />
          <Button buttonType={"cancel"} buttonText="Close" buttonMethod={"clearModal"} payload={""} />
        </div>
      </ModalContainer>
    );
  }
}
const mapStateToProps = state => ({
  modalButtonText: state.modalReducer.modalButtonText,
  modalHeader: state.modalReducer.modalHeader,
  modalButtonMethod: state.modalReducer.modalButtonMethod,
  activeModal: state.modalReducer.activeModal,
  editableRecipeIndex: state.modalReducer.editableRecipeIndex,
  recipes: state.recipeReducer.recipes
});
export default connect(mapStateToProps, actions)(Modal);
