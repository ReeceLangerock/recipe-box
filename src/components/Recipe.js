import React from 'react'
import styled from 'styled-components'
import Button from './Button'

const Container = styled.div`
  width: calc(100% * (1/4) - 20px - 1px);
  min-width: 200px;
  display: flex;
  flex-direction:column;
  justify-content: space-between;
  margin: 10px 5px 10px 5px;
  box-sizing: border-box;
  overflow: hidden;
  text-align: left;
  transition: all 0.75s ease-in;
  max-height: ${props => (props.expanded ? '400px' : '60px')};
  background-color: ${props => (props.expanded ? 'white' : 'white')};
  box-shadow: ${props => (props.expanded ? '0 10px 20px rgba(155,197,61, 0.29), 0 6px 6px rgba(155,197,61, 0.23)' : '0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23)')};
  border-radius: 3px;

  &:hover {
    transition: all 0.75s ease-out;

    box-shadow: 0 10px 20px rgba(155, 197, 61, 0.29), 0 6px 6px rgba(155, 197, 61, 0.23);
  }
  @media (max-width: 920px) {
    width: calc(100% * (1/3) - 10px - 2px);
  }

  @media (max-width: 700px) {
    width: calc(100% * (1/2) - 10px - 2px);
  }
  @media (max-width: 500px) {
    width: calc(100% - 10px - 1px);
  }
`

const Name = styled.div`
  font-size: 1.35rem;
  overflow: hidden;
  transition: all 0.75s ease-in-out;
  text-overflow: ellipsis;
  white-space: ${props => (props.expanded ? 'wrap' : 'nowrap')};
`

const IngredientsContainer = styled.div`
  border: 1px solid;
  border-radius: 2px;
  padding: 2px 0px 2px 0px;
  margin: 2px 5px 10px 5px;
  max-height: 320px;
  overflow-y: auto;
  flex-grow:2;
`
const Ingredient = styled.div`
  padding: 1px 4px 1px 4px;
  &:nth-child(even) {
    background: rgba(155, 197, 61, 0.15);
    border-bottom: 1px solid rgba(155, 197, 61, 1);
    border-top: 1px solid rgba(155, 197, 61, 1);
  }
`

const Header = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 17px 5px 17px 10px;
  margin-bottom: 10px;
  border-bottom: 1px solid;
  transition: all 0.75s ease-in-out;
  min-height: 61px;
  box-sizing: border-box;

  background-color: ${props => (props.expanded ? 'rgba(155,197,61,1)' : 'white')};
`

const Footer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 5px;
  padding-bottom: 5px;
  
`

export class Recipe extends React.Component {
  constructor (props) {
    super(props)
    this.handleExpand = this.handleExpand.bind(this)

    this.state = {
      expanded: false
    }
  }

  handleExpand () {
    this.setState({
      expanded: !this.state.expanded
    })
  }
  renderIngredients () {
    return this.props.ingredients.map((ingredient, index) => {
      return <Ingredient key={ingredient + index}>{ingredient}</Ingredient>
    })
  }
  render () {
    return (
      <Container expanded={this.state.expanded}>
        <Header expanded={this.state.expanded} onClick={this.handleExpand}>
          <Name expanded={this.state.expanded}>{this.props.name}</Name>
        </Header>

        <IngredientsContainer>{this.renderIngredients()}</IngredientsContainer>
        <Footer>
          <Button buttonType={'confirm'} buttonMethod={'setEditModal'} buttonText={'Edit'} payload={[this.props.index]} />
          <Button buttonType={'cancel'} buttonMethod={'deleteRecipe'} buttonText={'Delete'} payload={[this.props.index]} />
        </Footer>
      </Container>
    )
  }
}
export default Recipe
