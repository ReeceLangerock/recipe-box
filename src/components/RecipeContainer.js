import React from "react";
import styled from "styled-components";
import Recipe from "./Recipe";
import AddRecipe from "./AddRecipe";

import { connect } from "react-redux";

import * as actions from "../redux/actions/actions";

const Container = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: 10px;
  width: 95%;

  margin: 0 auto;
  justify-content: flex-start;
  position: relative;
  box-sizing: border-box;

`;

export class RecipeContainer extends React.Component {
  componentWillMount(){
    this.props.populateFromLocalStorage();
  }
  renderRecipeItems() {
    return this.props.recipes.map((recipe, index) => {
      return <Recipe key={`recipe${index}`} name={recipe.name} ingredients={recipe.ingredients} index={index} />;
    });
  }
  render() {
    return <Container><AddRecipe/>{this.renderRecipeItems()}</Container>;
  }
}

const mapStateToProps = state => ({
  recipes: state.recipeReducer.recipes
});
export default connect(mapStateToProps, actions)(RecipeContainer);
