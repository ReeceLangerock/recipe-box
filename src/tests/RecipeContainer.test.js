import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, find, render, simulate } from "enzyme";
import enzymeSerializer from "enzyme-to-json/serializer";
import { spy } from "sinon";
import { RecipeContainer } from "./../components/RecipeContainer";
import configureStore from "redux-mock-store";
import Recipe from "../components/Recipe";

const initialState = {};
let mockStore = configureStore(initialState);
expect.addSnapshotSerializer(enzymeSerializer);
describe("RecipeContainer", () => {


  let component;
  let props;
  beforeEach(() => {
    props = { recipes: [{ name: "Cookies", id: 1, ingredients: ["sugar"] }, 
    { name: "Pizza", id: 2, ingredients: ["cheese"] }] };
    component = shallow(<RecipeContainer store={mockStore} {...props} />);
  });

  it("renders each recipe", () => {
    expect(component.find(Recipe).length).toEqual(props.recipes.length);
  });

  it("matches snapshot", () => {
    const comp = shallow(<RecipeContainer {...props} />);
    expect(comp).toMatchSnapshot();
  });
});
