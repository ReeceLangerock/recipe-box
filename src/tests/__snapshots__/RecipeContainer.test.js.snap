// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeContainer matches snapshot 1`] = `
<styled.div>
  <Recipe
    id={1}
    ingredients={
      Array [
        "sugar",
      ]
    }
    name="Cookies"
  />
  <Recipe
    id={2}
    ingredients={
      Array [
        "cheese",
      ]
    }
    name="Pizza"
  />
</styled.div>
`;
